File: SeungyeonJi/week7/2023-10-29-SumRootToLeafNumbers.js
var sumNumbers = function (root) {
  if (!root) null;
  let result = [];

  function recur(root, curArr) {
    if (!root) null;
    let curPit = root;
    curArr.push(curPit.val);

    if (!root.left && !root.right) {
      let num = Number(curArr.join(""));
      result.push(num);
    }

    if (root.left) recur(root.left, curArr);
    if (root.right) recur(root.right, curArr);
    curArr.pop();
  }
  recur(root, []);
  let num = 0;
  for (let i = 0; i < result.length; i++) {
    num += result[i];
  }

  return num;
};

File: TaylorJi/w7/2023-10-29-Sum-Root-To-Leaf-Numbers.java
class Solution {
    int result=0;
    public void sum(TreeNode root, int num) {
        if(root == null) return;
        num = root.val + ( 10 * num);

        if(root.left == null && root.right == null){
            result += num;
        }
        sum(root.left, num);
        sum(root.right, num);
    }
    public int sumNumbers(TreeNode root) {
        sum(root, 0);
        return result;
    }
}